lms/djangoapps/discussion/views.py

```diff
@@ -361,9 +376,13 @@ def single_thread(request, course_key, discussion_id, thread_id):
             'annotated_content_info': annotated_content_info,
         })
     else:
-        course_id = str(course.id)
-        tab_view = CourseTabView()
-        return tab_view.get(request, course_id, 'discussion', discussion_id=discussion_id, thread_id=thread_id)
+        redirect_url = _get_redirect_url_to_new_mfe(request, str(course.id), thread_id)
+        if redirect_url is None:
```

Comment:
This condition can be simplified by an early return
```
if redirect_url:
     return redirect(redirect_url) 
```
Also redirect_url_to_new_mfe will make it more clear

Reply:
The redirect will only work if a single thread url is accessed?